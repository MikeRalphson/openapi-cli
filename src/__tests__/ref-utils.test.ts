import { parseRef } from '../ref-utils';

describe('ref-utils', () => {
  it(`should unescape refs with '/'`, () => {
    const reference = 'somefile.yaml#/components/schemas/scope~1domain-schema';
    expect(parseRef(reference)).toMatchInlineSnapshot(`
      Object {
        "pointer": Array [
          "components",
          "schemas",
          "scope/domain-schema",
        ],
        "uri": "somefile.yaml",
      }
    `);
  });

  it(`should unescape refs with '~'`, () => {
    const reference = 'somefile.yaml#/components/schemas/complex~0name';
    expect(parseRef(reference)).toMatchInlineSnapshot(`
      Object {
        "pointer": Array [
          "components",
          "schemas",
          "complex~name",
        ],
        "uri": "somefile.yaml",
      }
    `);
  });

  it(`should unescape complex urlencoded paths`, () => {
    const referene = 'somefile.yaml#/components/schemas/scope%2Fcomplex~name';
    expect(parseRef(referene)).toMatchInlineSnapshot(`
      Object {
        "pointer": Array [
          "components",
          "schemas",
          "scope/complex~name",
        ],
        "uri": "somefile.yaml",
      }
    `);
  });
});
