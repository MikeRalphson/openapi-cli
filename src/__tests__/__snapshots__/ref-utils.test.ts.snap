// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ref-utils should validate definition with urlencoded paths 1`] = `Array []`;
